pmd-cli/src/main/java/net/sourceforge/pmd/cli/commands/internal/AbstractAnalysisPmdSubcommand.java

```diff
@@ -99,6 +127,30 @@ protected final void validate() throws ParameterException {
         }
     }
 
+    protected void configureCommonOptions(C configuration) {
+
+        // Setup CLI message reporter
+        configuration.setReporter(new SimpleMessageReporter(LoggerFactory.getLogger(PmdCommand.class)));
+
+        if (inputPaths != null) {
+            configuration.setInputPathList(new ArrayList<>(inputPaths));
+        }
+        configuration.setInputFilePath(fileListPath);
+        configuration.setIgnoreFilePath(ignoreListPath);
+        configuration.setInputUri(uri);
+        configuration.setSourceEncoding(encoding.getEncoding());
+        if (relativizeRootPaths != null) {
+            configuration.addRelativizeRoots(relativizeRootPaths);
+        }
+        for (LanguageFilePattern pat : languageFilePatterns) {
+            Language lang = configuration.getLanguageRegistry().getLanguageById(pat.getLanguageid());
+            if (lang == null) {
+                configuration.getReporter().warn("Language {0} mentioned in --assign-language option is not loaded.", pat.getLanguageid());
```

Comment:
We can't know in the CLI :/

Reply:
Should this be a warning or a `ParameterException`? Actually, why is this not validated directly in `LanguageFilePatternConverter`? Rather than have `langId` be a `String` use the `Language` and have it strongly typed in `LanguageFilePattern`. At the end of the day, te validation the `Language` exists should still be needed regardless of the entry point (CLI in this case). Requiring a non-null `Language` instance to create the `LanguageFilePattern` enforces that throughout.

Also, I'd speak of a language being "unknown" instead of loaded / not loaded. This would be consistent with what we currently throw when we set an invalid language elsewhere.

Reply:
Maybe a ParameterException yes...

I thought it would be useful to be able to define patterns even for languages that are not loaded... The use case I had in mind is adding patterns for POM even if it is not loaded, so that XML does not take precedence. If we make languages register their patterns upon loading then that should not be a valid use case anymore.